chainercv/functions/psroi_pooling_2d.py

```diff
@@ -0,0 +1,331 @@
+            float roi_height = max(roi_end_h - roi_start_h, 0.1);
+
+            // Compute w and h at bottom
+            float bin_size_w = roi_width / static_cast<float>(pooled_width);
+            float bin_size_h = roi_height / static_cast<float>(pooled_height);
+
+            int wstart = floor(
+                static_cast<float>(pw) * bin_size_w + roi_start_w);
+            int hstart = floor(
+                static_cast<float>(ph) * bin_size_h + roi_start_h);
+            int wend = ceil(
+                static_cast<float>(pw + 1.0) * bin_size_w + roi_start_w);
+            int hend = ceil(
+                static_cast<float>(ph + 1.0) * bin_size_h + roi_start_h);
+
+            // Add roi offsets and clip to input boundaries
+            wstart = min(max(wstart, 0), width);
+            hstart = min(max(hstart, 0), height);
+            wend = min(max(wend, 0), width);
+            hend = min(max(hend, 0), height);
+            bool is_empty = (hend <= hstart) || (wend <= wstart);
+
+            // Compute c at bottom
+            int gw = floor(
+                static_cast<float>(pw) * group_size / pooled_width);
+            int gh = floor(
+                static_cast<float>(ph) * group_size / pooled_height);
+            gw = min(max(gw, 0), group_size - 1);
+            gh = min(max(gh, 0), group_size - 1);
+            int c = (ctop * group_size + gh) * group_size + gw;
+
+            int bottom_diff_offset = (roi_batch_ind * channels + c);
+            bottom_diff_offset = bottom_diff_offset * height * width;
+            float bin_area = (hend - hstart) * (wend - wstart);
+            float diff_val = is_empty ? (float) 0. : top_diff / bin_area;
+            for (int h = hstart; h < hend; ++h){
+              for (int w = wstart; w < wend; ++w){
+                int bottom_index = h * width + w;
+                atomicAdd(
+                    &bottom_diff[bottom_diff_offset + bottom_index], diff_val);
+              }
+            }
+            ''', 'psroi_pooling_2d_bwd'
+        )(bottom_diff, bottom_rois.shape[0], self.spatial_scale,
+          channels, height, width, self.outd, self.outh, self.outw,
+          self.group_size, bottom_rois, gy[0])
+
+        return bottom_diff, None
+
+
+def psroi_pooling_2d(
+        x, rois, outd, outh, outw, spatial_scale,
+        group_size
+):
+    """Position Sensitive Region of Interest (ROI) pooling function.
+
+    This function computes position sensitive average of input spatial patch
+    with the given region of interest. Each ROI are splitted into
+    :math:`(group\_size, group\_size)` regions, and position sensitive values
+    in each region is inferenced.
+
+    Args:
+        x (~chainer.Variable): Input variable. The shape is expected to be
+            4 dimentional: (n: batch, c: channel, h, height, w: width).
+        rois (~chainer.Variable): Input roi variable. The shape is expected to
```

Comment:
I think it is not necessary to make the API of this function to follow `chainer.functions.roi_pooling_2d`.
Rather, it would be better to follow the preferred interface of ChainerCV.

Can you change the second input to `rois` and `roi_indices` as defined here? (`bbox` is `yx` order)
```
            rois (array): A bounding box array containing coordinates of
                proposal boxes.  This is a concatenation of bounding box
                arrays from multiple images in the batch.
                Its shape is :math:`(R', 4)`. Given :math:`R_i` proposed
                RoIs from the :math:`i` th image,
                :math:`R' = \\sum _{i=1} ^ N R_i`.
            roi_indices (array): An array containing indices of images to
                which bounding boxes correspond to. Its shape is :math:`(R',)`.
```
https://github.com/chainer/chainercv/blob/master/chainercv/links/model/faster_rcnn/faster_rcnn_vgg.py#L201

Reply:
This would require some change to kernels.